src/main/java/com/minecolonies/coremod/colony/jobs/JobDeliveryman.java

```diff
@@ -87,61 +131,104 @@ public SoundEvent getMoveAwaySound()
         return null;
     }
 
-    @Override
-    public SoundEvent getBedTimeSound()
+    /**
+     * Returns whether or not the job has a currentTask.
+     *
+     * @return true if has currentTask, otherwise false.
+     */
+    public boolean hasTask()
     {
-        if (getCitizen() != null)
+        return !taskQueue.isEmpty() || returning;
+    }
+
+    /**
+     * Returns the {@link IRequest} of the current Task.
+     *
+     * @return {@link IRequest} of the current Task.
+     */
+    public IRequest<? extends Delivery> getCurrentTask()
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)